src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -317,17 +317,7 @@ public char PeekChar(int delta)
             int position = this.Position;
             this.AdvanceChar(delta);
 
-            char ch;
-            if (_offset >= _characterWindowCount
-                && !MoreChars())
-            {
-                ch = InvalidCharacter;
-            }
-            else
-            {
-                // N.B. MoreChars may update the offset.
-                ch = _characterWindow[_offset];
-            }
+            char ch = PeekChar();
```

Comment:
>char ch = PeekChar(); [](start = 12, length = 21)

This change looks unrelated to the feature. Consider moving it to a separate PR and merging it to master.